packages/landing/src/components/map.tsx

```diff
@@ -44,6 +45,16 @@ export class Basemaps extends Component<unknown, { isLayerSwitcherEnabled: boole
     if (location.pitch != null) this.map.setPitch(location.pitch);
   };
 
+  updateTerrainFromEvent = (): void => {
+    const terrain = this.map.getTerrain();
+    if (terrain) {
+      Config.map.terrain = terrain.source;
+    } else {
+      Config.map.terrain = null;
+    }
+    window.history.pushState(null, '', `?${MapConfig.toUrl(Config.map)}`);
```

Comment:
do we need to validate that this is actually a change? `Config.map.terrain != terrain.source`